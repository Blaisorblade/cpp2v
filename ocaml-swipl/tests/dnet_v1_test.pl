/* Making sure we [exit 1] on a user-error. */
set_prolog_flag(on_error, halt).

/* Loading the file with our rules. */
[dnet_v1].

/* Adding the premises. */
assertz(prem(0, 'P')).
assertz(prem(1, 'Q')).
assertz(prem(2, 'R')).
assertz(prem(3, 'S')).
assertz(prem(4, 'T')).

/* Adding the conclusions. */
assertz(cncl(0, 'A')).
assertz(cncl(1, 'Q')).
assertz(cncl(2, 'B')).
assertz(cncl(3, 'P')).
assertz(cncl(4, 'C')).

/* Adding cancel hints. */
assertz(cancel(0, [X], [X])).

/* Query. */
call_nth(
  (match_any(KIND, HID, PIDS, CIDS),
   format("KIND=~w, HID=~w, PIDS=~w, CIDS=~w~n", [KIND, HID, PIDS, CIDS])),
  100
).
