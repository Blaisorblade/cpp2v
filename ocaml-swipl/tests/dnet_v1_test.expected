true.

true.

true.

true.

true.

true.

true.

true.

true.

true.

true.

true.

true.

KIND=cancel, HID=0, PIDS=[0], CIDS=[3]
KIND=cancel, HID=0, PIDS=[1], CIDS=[1]
false.


